### PR TITLE
feat(tooltip): #MC-206 add last update in external calendar tooltip

### PR DESCRIPTION
**Pull request by alicedraillard**
## Describe your changes
add tooltip that displays external calendar's last update

## Checklist tests
place mouse over external calendar in sidebar and check tooltip

## Issue ticket number and link
#MC-206 https://jira.support-ent.fr/browse/MC-206

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by sjollois — APPROVED** [review]

File: src/main/resources/public/ts/directives/calendar-item/calendar-item.ts [hunk 1]
```diff
@@ -1,14 +1,16 @@
-import {model, ng} from "entcore";
+import {model, moment, ng} from "entcore";
 import {ROOTS} from "../../core/const/roots";
 import {Calendar} from "../../model";
 import {IScope} from "angular";
 import {Subject} from "rxjs";
 import {ICalendarService} from "../../services";
+import {idiom as lang} from "entcore/types/src/ts/idiom";
```
Comment (JinDouang) [inline src/main/resources/public/ts/directives/calendar-item/calendar-item.ts:7]: Tu n'as pas l'air de l'utiliser

File: src/main/resources/public/ts/model/Calendar.ts [hunk 2]
```diff
@@ -18,6 +20,7 @@ export class Calendar implements Selectable, Shareable {
     isExternal: boolean;
     icsLink: string;
     icsLinkPlatform: string;
+    updated: object;
```
Comment (JinDouang) [inline src/main/resources/public/ts/model/Calendar.ts:23]: d'où viennent ces fields que tu as rajouté ? "created", "modified" et "updated"

File: src/main/resources/public/ts/directives/calendar-item/calendar-item.ts [hunk 3]
```diff
@@ -44,6 +46,10 @@ class Controller implements ng.IController, IViewModel {
     updateExternalCalendar = async (): Promise<void> => {
         await this.calendarService.updateExternalCalendar(this.$scope.vm.calendar);
     };
+
+    getLastUpdate = (format: string): string => {
+        return moment(this.$scope.vm.calendar.updated['$date']).format(format);
```
Comment (JinDouang) [inline src/main/resources/public/ts/directives/calendar-item/calendar-item.ts:51]: d'où vient ce scope "updated" au niveau du controller ?

**Review by JinDouang — CHANGES_REQUESTED** [review]

File: src/main/resources/public/ts/directives/calendar-item/calendar-item.ts (hunk 3 above)
Comment (alicedraillard) [reply src/main/resources/public/ts/directives/calendar-item/calendar-item.ts:51]: c'est un nouveau champ qui sera conservé en bdd (comme modified et created qui existent actuellement)

Comment (JinDouang) [reply src/main/resources/public/ts/directives/calendar-item/calendar-item.ts:49]: vu avec @alicedraillard pour adapter le model Calendar

File: src/main/resources/public/ts/model/Calendar.ts (hunk 2 above)
Comment (JinDouang) [reply src/main/resources/public/ts/model/Calendar.ts:23]: vu avec @alicedraillard pour adapter le model Calendar

**Review by JinDouang — CHANGES_REQUESTED** [review]

File: src/main/resources/public/ts/model/Calendar.ts [hunk 4]
```diff
@@ -26,6 +30,12 @@ export class Calendar implements Selectable, Shareable {
         if (!_.isEmpty(calendar)) {
             this.myRights.fromBehaviours();
             Mix.extend(this, Behaviours.applicationsBehaviours.calendar.resourceRights(calendar));
+            // calendar.updated['$date'] is a number (timestamp)
+            if (calendar.updated && calendar.updated['$date']) {
+                console.log(calendar.updated['$date'].toString());
+                console.log(FORMAT.formattedISODate);
```
Comment (JinDouang) [inline src/main/resources/public/ts/model/Calendar.ts:36]: console.log

**Review by JinDouang — CHANGES_REQUESTED** [review]

**Review by JinDouang — APPROVED** [review]

**Pull request merged**